pkg/spec/deploymentconfig.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/ghodss/yaml"
+	os_deploy_v1 "github.com/openshift/origin/pkg/deploy/apis/apps/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kapi "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// Unmarshal the Kedge YAML file
+func (deployment *DeploymentConfigSpecMod) Unmarshal(data []byte) error {
+	err := yaml.Unmarshal(data, &deployment)
+	if err != nil {
+		return errors.Wrap(err, "could not unmarshal into internal struct")
+	}
+	log.Debugf("object unmarshalled: %#v\n", deployment)
+	return nil
+}
+
+// Validate all portions of the file
+func (deployment *DeploymentConfigSpecMod) Validate() error {
+
+	if err := validateVolumeClaims(deployment.VolumeClaims); err != nil {
+		return errors.Wrap(err, "error validating volume claims")
+	}
+
+	return nil
+}
+
+// Fix all services / volume claims / configmaps that are applied
+// TODO: abstract out this code when more controllers are added
+func (deployment *DeploymentConfigSpecMod) Fix() error {
+
+	var err error
+
+	// fix deployment.Services
+	deployment.Services, err = fixServices(deployment.Services, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "Unable to fix services")
+	}
+
+	// fix deployment.VolumeClaims
+	deployment.VolumeClaims, err = fixVolumeClaims(deployment.VolumeClaims, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "Unable to fix persistentVolume")
+	}
+
+	// fix deployment.configMaps
+	deployment.ConfigMaps, err = fixConfigMaps(deployment.ConfigMaps, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix configMaps")
+	}
+
+	deployment.Containers, err = fixContainers(deployment.Containers, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix containers")
+	}
+
+	deployment.InitContainers, err = fixContainers(deployment.InitContainers, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix init-containers")
+	}
+
+	deployment.Secrets, err = fixSecrets(deployment.Secrets, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix secrets")
+	}
+
+	return nil
+}
+
+func (deployment *DeploymentConfigSpecMod) Transform() ([]runtime.Object, []string, error) {
```

Comment:
Can we structure/refactor this to the same way the other `Transform()` methods are?
Basically the order in which the functions are called, like first CreateK8sObjects, then CreateController, then a couple of checks, then GetScheme and GetGVK?